Fc 732 hipay customdata (#324)
front-commerce/developers.front-commerce.com: source/docs/advanced/payments/hipay.md

<blockquote class="note">
Please keep in mind that HiPay payment methods depends on the Customer country and the Store currency. Different contexts could display different payment methods in the checkout.
</blockquote>

### Advanced: customize data sent to HiPay

The HiPay payment module is extensible. It leverages Front-Commerce's "data transform" pattern to allow developers to customize payment data sent to HiPay.

You need to create a backend loader to register the transformation.

```
export default {
  namespace: "HiPay/Custom",
  dependencies: ["HiPay/Core"],
  contextEnhancer: ({ loaders }) => {

>     this.loaders.HiPay.registerPaymentDataTransform((customer, order) => ({

Reviewer: ```suggestion
    loaders.HiPay.registerPaymentDataTransform((customer, order) => ({
```

<blockquote class="note">
Please keep in mind that HiPay payment methods depends on the Customer country and the Store currency. Different contexts could display different payment methods in the checkout.
</blockquote>

### Advanced: customize data sent to HiPay

> The HiPay payment module is extensible. It leverages Front-Commerce's "data transform" pattern to allow developers to customize payment data sent to HiPay.

Reviewer 2: ```suggestion
The HiPay payment module is extensible. It leverages the "data transform" pattern from Front-Commerce to allow developers to customize payment data sent to HiPay.
```

1. Override the HiPayCheckout to gather relevant data

```
mkdir -p my-module/web/theme/hipay
cp -u node_modules/front-commerce/modules/payment-hipay/web/theme/hipay/HiPayCheckout.js my-module/web/theme/hipay/HiPayCheckout.js
```

```diff
...
  const onSubmit = async () => {
    if (!submitData) {
      return;
    }
    onAuthorize({
      stateData: JSON.stringify({
        ...submitData,
        customer,
+       customData: []
      }),
    });
  };
...
```

2. Create a backend loader to register the transformation

```
export default {
  namespace: "HiPay/Custom",
  dependencies: ["HiPay/Core"],
  contextEnhancer: ({ loaders }) => {
    this.loaders.HiPay.registerPaymentDataTransform((paymentData) => ({
      custom_data: JSON.stringify(paymentData.customData)
    }))
  },
};
```

<blockquote class="warning">

> 

Reviewer 2: **non-blocking :** remove additional line

```suggestion
```

**Warning:** root attributes like `custom_data` indicated here are directly added to the HiPay `/order` form-data ([see the webservice documentation](https://developer.hipay.com/api-explorer/api-online-payments#/payments/requestNewOrder)). Add other attributes cautiously.

> 

Reviewer 2: **non-blocking :** remove additional line

```suggestion
```
